cli/cli/cmds_volume.go

```diff
@@ -339,7 +344,11 @@ func (c *CLI) initVolumeCmds() {
 			// volumes that are attached or can be attached should be valid
 			// candidates for a mount operation
 			result, err := c.lsVolumes(
-				args, volumeStatusAttached, volumeStatusAvailable)
+				args,
+				requestAttachments(
+					apitypes.VolumeAttachmentsMine,
+					apitypes.VolumeAttachmentsAttached,
+					apitypes.VolumeAttachmentsUnattached))
```

Comment:
Hi @codenrhoden,

So I hear you and I understand. I think part of the issue is that I want to be able to do this, that, and the other with a single query. Meaning I'd like to be able to say "get all volumes and their attachments" or "get all volumes that are attached" or "get all volumes that are not attached" or "get all volumes that are attached to me and all volumes that are not attached".

What I don't want is for people to have to worry about is performing multiple queries to get what they want. Let's review the values here:

```
const (
	// VolumeAttachmentsRequested indicates attachment information is requested.
	VolumeAttachmentsRequested VolumeAttachmentsTypes = 1 << iota // 1

	// VolumeAttachmentsMine indicates attachment information should
	// be returned for volumes attached to the instance specified in the
	// instance ID request header. If this bit is set then the instance ID
	// header is required.
	VolumeAttachmentsMine // 2

	// VolumeAttachmentsDevices indicates an attempt should made to map devices
	// provided via the local devices request header to the appropriate
	// attachment information. If this bit is set then the instance ID and
	// local device headers are required.
	VolumeAttachmentsDevices // 4

	// VolumeAttachmentsAttached indicates only volumes that are attached
	// should be returned.
	VolumeAttachmentsAttached // 8

	// VolumeAttachmentsUnattached indicates only volumes that are unattached
	// should be returned.
	VolumeAttachmentsUnattached // 16
)
```

As I see it, here are the valid masks:

Mask | Components | Description
------|--------------|------------
`0`| `0` | The default value. This indicates no attachment information is requested.
`1` | `1` | Request attachment information for all retrieved volumes
`3` | `1|2` | Request attachment information for volumes attached to the instance provided in the instance ID
`7` | `1|2|4` | Request attachment information for volumes attached to the instance provided in the instance ID and perform device mappings where possible.
`9` | `1|8` | Request attachment information for all retrieved volumes and return only volumes that are attached to some instance.
`17` | `1|16` | Request attachment information for all retrieved volumes and return only volumes that are not attached to any instance.

Here's where it gets a little more interesting:

Mask | Components | Description
------|--------------|------------
`11` | `1|2|8` | Request attachment information for all retrieved volumes and return only volumes that attached to the instance provided in the instance ID.
`27` | `1|2|8|16` | Request attachment information for all retrieved volumes and return only volumes that attached to the instance provided in the instance ID *or* are not attached to any instance at all. tl;dr - _Attached To Me or Available_